src/main/java/mekanism/common/registries/MekanismRobitSkins.java

```diff
@@ -13,7 +18,14 @@ private MekanismRobitSkins() {
     public static final RobitSkinDeferredRegister ROBIT_SKINS = new RobitSkinDeferredRegister(Mekanism.MODID);
 
     public static final RobitSkinRegistryObject<RobitSkin> BASE = ROBIT_SKINS.register("robit");
-    public static final RobitSkinRegistryObject<RobitSkin> LESBIAN = ROBIT_SKINS.register("lesbian", 8);
-    public static final RobitSkinRegistryObject<RobitSkin> PRIDE = ROBIT_SKINS.register("pride", 6);
-    public static final RobitSkinRegistryObject<RobitSkin> TRANS = ROBIT_SKINS.register("trans", 4);
+
+    public static final Map<RobitPrideSkinData, RobitSkinRegistryObject<RobitSkin>> PRIDE_SKINS;
+
+    static {
+        Map<RobitPrideSkinData, RobitSkinRegistryObject<RobitSkin>> tempMap = new HashMap<>();
+        for (RobitPrideSkinData data: RobitPrideSkinData.values()) {
+            tempMap.put(data, ROBIT_SKINS.register(data.lowerCaseName(), data.getColor().length));
+        }
+        PRIDE_SKINS = ImmutableMap.copyOf(tempMap);
+    }
```

Comment:
To ensure this maintains order between runs and then gives the registry id it expects I feel like this should instead be:
```java
public static final Map<RobitPrideSkinData, RobitSkinRegistryObject<RobitSkin>> PRIDE_SKINS = Util.make(() -> {
    Map<RobitPrideSkinData, RobitSkinRegistryObject<RobitSkin>> internal = new EnumMap<>(RobitPrideSkinData.class);
    for (RobitPrideSkinData data: RobitPrideSkinData.values()) {
        internal .put(data, ROBIT_SKINS.register(data.lowerCaseName(), data.getColor().length));
    }
    return Collections.unmodifiableMap(internal);
});
```